addons/product/models/product_template.py

```diff
@@ -179,70 +179,84 @@ def _compute_currency_id(self):
-                template.product_variant_ids.standard_price = template.standard_price
+        self._set_product_variant_field('standard_price')
 
     def _search_standard_price(self, operator, value):
-        products = self.env['product.product'].search([('standard_price', operator, value)], limit=None)
-        return [('id', 'in', products.mapped('product_tmpl_id').ids)]
+        return [('product_variant_ids.standard_price', operator, value)]
 
-    @api.depends('product_variant_ids', 'product_variant_ids.volume')
+    @api.depends('product_variant_ids.volume')
     def _compute_volume(self):
-        unique_variants = self.filtered(lambda template: len(template.product_variant_ids) == 1)
-        for template in unique_variants:
-            template.volume = template.product_variant_ids.volume
-        for template in (self - unique_variants):
-            template.volume = 0.0
+        self._compute_template_field_from_variant_field('volume')
 
     def _set_volume(self):
-        for template in self:
-            if len(template.product_variant_ids) == 1:
-                template.product_variant_ids.volume = template.volume
+        self._set_product_variant_field('volume')
 
-    @api.depends('product_variant_ids', 'product_variant_ids.weight')
+    @api.depends('product_variant_ids.weight')
     def _compute_weight(self):
-        unique_variants = self.filtered(lambda template: len(template.product_variant_ids) == 1)
-        for template in unique_variants:
-            template.weight = template.product_variant_ids.weight
-        for template in (self - unique_variants):
-            template.weight = 0.0
+        self._compute_template_field_from_variant_field('weight')
+
+    def _set_weight(self):
+        self._set_product_variant_field('weight')
 
     def _compute_is_product_variant(self):
         self.is_product_variant = False
 
     @api.depends('product_variant_ids.barcode')
     def _compute_barcode(self):
-        self.barcode = False
-        for template in self:
-            # TODO master: update product_variant_count depends and use it instead
-            variant_count = len(template.product_variant_ids)
-            if variant_count == 1:
-                template.barcode = template.product_variant_ids.barcode
-            elif variant_count == 0:
-                archived_variants = template.with_context(active_test=False).product_variant_ids
-                if len(archived_variants) == 1:
-                    template.barcode = archived_variants.barcode
+        self._compute_template_field_from_variant_field('barcode')
 
     def _search_barcode(self, operator, value):
-        templates = self.with_context(active_test=False).search([('product_variant_ids.barcode', operator, value)])
-        return [('id', 'in', templates.ids)]
+        subquery = self.with_context(active_test=False)._search(
+            [('product_variant_ids.barcode', operator, value)]
+        )
```

Comment:
```suggestion
        subquery = self.with_context(active_test=False)._search([
            ('product_variant_ids.barcode', operator, value),
        ])
```
we tend to keep one domain leaf for each line when we indent search calls.
That way, when we add a new leaf, there is only one modified line in the git diff :).